src/test/javascript/spec/component/exam/participate/summary/points-summary/grade-key-overview/grade-key-overview.component.spec.ts

```diff
@@ -0,0 +1,103 @@
+import * as sinon from 'sinon';
+import { GradingSystemService } from 'app/grading-system/grading-system.service';
+import { GradeKeyOverviewComponent } from 'app/exam/participate/summary/points-summary/grade-key-overview/grade-key-overview.component';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
+import { MockComponent, MockDirective, MockModule, MockPipe, MockProvider } from 'ng-mocks';
+import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
+import { ActivatedRoute, Router } from '@angular/router';
+import { JhiTranslateDirective } from 'ng-jhipster';
+import { of } from 'rxjs';
+import { FaIconComponent } from '@fortawesome/angular-fontawesome';
+import { MockRouter } from '../../../../../../helpers/mocks/service/mock-route.service';
+import { GradeStep, GradeStepsDTO } from 'app/entities/grade-step.model';
+import { GradeType } from 'app/entities/grading-scale.model';
+import { HttpResponse } from '@angular/common/http';
+
+describe('GradeKeyOverviewComponent', () => {
+    let fixture: ComponentFixture<GradeKeyOverviewComponent>;
+    let comp: GradeKeyOverviewComponent;
+
+    let gradingSystemService: GradingSystemService;
+    let router: Router;
+
+    const gradeStep1: GradeStep = {
+        gradeName: 'Fail',
+        lowerBoundPercentage: 0,
+        upperBoundPercentage: 50,
+        lowerBoundInclusive: true,
+        upperBoundInclusive: false,
+        isPassingGrade: false,
+    };
+    const gradeStep2: GradeStep = {
+        gradeName: 'Pass',
+        lowerBoundPercentage: 50,
+        upperBoundPercentage: 100,
+        lowerBoundInclusive: true,
+        upperBoundInclusive: true,
+        isPassingGrade: true,
+    };
+    const gradeStepsDto: GradeStepsDTO = {
+        examTitle: 'Title',
+        gradeType: GradeType.BONUS,
+        gradeSteps: [gradeStep1, gradeStep2],
+    };
+
+    beforeEach(() => {
+        return TestBed.configureTestingModule({
+            imports: [MockModule(NgbModule)],
+            declarations: [GradeKeyOverviewComponent, MockComponent(FaIconComponent), MockPipe(ArtemisTranslatePipe), MockDirective(JhiTranslateDirective)],
+            providers: [
+                { provide: ActivatedRoute, useValue: { params: of({ courseId: 1, examId: 1 }), queryParams: of({ grade: '2.0' }) } },
```

Comment:
Can you give different numbers to course and exam id?

Reply:
Will do